modules/@apostrophecms/area/index.js

```diff
@@ -144,6 +163,56 @@ module.exports = {
           canEdit
         });
       },
+      // Replace documents' area objects with rendered HTML for each area.
+      // This is used by GET requests including the `renderareas` query
+      // parameter. `within` is an array of Apostrophe documents.
+      async renderDocsAreas(req, within) {
+        within = Array.isArray(within) ? within : [];
+        let index = 0;
+        // Loop over the docs in the array passed in.
+        for (const doc of within) {
+          const rendered = [];
+
+          const areasToRender = {};
+          // Walk the document's areas and stash the areas for rendering later.
+          self.walk(doc, async function (area, dotPath) {
+            if (rendered.findIndex(path => dotPath.startsWith(path)) > -1) {
+              return;
+            }
+            // We're only rendering areas on the document, not ancestor or
+            // child page documents.
+            const regex = /^_(ancestors|children)|\._(ancestors|children)/;
+            if (dotPath.match(regex)) {
+              return;
+            }
+            const pathSplit = dotPath.split('.');
```

Comment:
Was there a cleaner way to get the parent?

Reply:
Nope that's the one.